modules/core/src/main/scala/org/tessellation/config/types.scala

```diff
@@ -53,35 +50,22 @@ object types {
     inMemoryCapacity: NonNegLong
   )
 
-  case class SoftStakingAndTestnetConfig(
-    softStakeAddress: Address = Address("DAG77VVVRvdZiYxZ2hCtkHz68h85ApT5b2xzdTkn"),
-    testnetAddress: Address = Address("DAG0qE5tkz6cMUD5M2dkqgfV4TQCzUUdAP5MFM9P"),
-    startingOrdinal: EpochProgress = EpochProgress(0L),
-    testnetCount: NonNegLong = 75L,
-    testnetWeight: NonNegLong = 4L,
-    softStakeCount: NonNegLong = 5218L,
-    softStakeWeight: NonNegLong = 4L,
-    facilitatorWeight: NonNegLong = 6L
-  )
-
-  case class DTMConfig(
-    address: Address = Address("DAG0Njmo6JZ3FhkLsipJSppepUHPuTXcSifARfvK"),
-    dtmWeight: NonNegLong = 195L,
-    remainingWeight: NonNegLong = 805L
-  )
+  @newtype
+  case class Weight(value: NonNegLong)
 
-  case class StardustConfig(
-    addressPrimary: Address = Address("DAGSTARDUSTCOLLECTIVEHZOIPHXZUBFGNXWJETZVSPAPAHMLXS"),
-    addressSecondary: Address = Address("DAG8VT7bxjs1XXBAzJGYJDaeyNxuThikHeUTp9XY"),
-    primaryWeight: NonNegLong = 1L,
-    secondaryWeight: NonNegLong = 1L,
-    remainingWeight: NonNegLong = 18L
+  case class ProgramsDistributionConfig(
+    weights: Map[Address, Weight] = Map(
+      Address("DAGSTARDUSTCOLLECTIVEHZOIPHXZUBFGNXWJETZVSPAPAHMLXS") -> Weight(5L), // stardust tax primary
+      Address("DAG8VT7bxjs1XXBAzJGYJDaeyNxuThikHeUTp9XY") -> Weight(5L), // stardust tax secondary
+      Address("DAG77VVVRvdZiYxZ2hCtkHz68h85ApT5b2xzdTkn") -> Weight(20L), // soft staking
+      Address("DAG0qE5tkz6cMUD5M2dkqgfV4TQCzUUdAP5MFM9P") -> Weight(1L), // testnet
+      Address("DAG0Njmo6JZ3FhkLsipJSppepUHPuTXcSifARfvK") -> Weight(65L) // data pool
+    ),
+    remainingWeight: Weight = Weight(4L) // facilitators
```

Comment:
What do you think about calculating it as `100L - sum(weights)`? We can also add a unit test which checks if all the weights sum up to `Weight(100L)`.

Reply:
Actually, that's a good idea. This test may have to be amended in the future because weights don't have to sum up to 100. It would be perfectly fine to have 3 weights of 1, so the sum would be 3 (not 100) and rewards would be distributed like this `{ 1/3, 1/3, 1/3 }`. But this test would help anyway if someone would unintentionally change the sum to different than 100, so I think it's worth adding it.

Reply:
>This test may have to be amended in the future because weights don't have to sum up to 100

Makes sense. So maybe just checking if `<= 100`?

Reply:
I could also add the `exptectedWeightSum` to the config and validate it during the application start. What do you think about it? 🤔

Reply:
Someone can still put wrong value in `expectedWeightSum` 🤔 . But assuming that expected one is set correctly then it is still better than nothing so fine for me. I just can't make my mind if `expectedWeightSum` should be in config or in unit test.

Reply:
I've added verification for `expextedWeightSum` only in the unit test for now, since rewards configuration isn't dynamically loaded

Reply:
Unit test is fine for me too